TutorialApplication/OrganizationInfoViewController.swift

```diff
@@ -14,6 +14,87 @@ import UIKit
     
     static let kEmployeesOrderHasChanged = "EmployeesOrderHasChanged"
     
+    static let kOrganizationHasChanged = "OrganizationHasChanged"
+    
+    @IBAction func fetchOrganizations()
+    {
+        RequestManager.fetchOrganizations { (response) in
+        
+        var newOrganization:FFOrganization
+        
+        var newEmployee:FFEmployee
+            
+        let countOfOrganizations = DatabaseController.requestResults(for: nil, sortDescriptors: nil, entity: "FFOrganization")
+        print(countOfOrganizations?.count as Any)
+        
+        if (countOfOrganizations?.count)! <= 1
+        {
+        let organizations = response["organizations"]
+        
+        for jsonObjects in organizations as! [AnyObject]
+            {
+                let organizationName = jsonObjects["name"] as! String
+                
+                let organizationEmployees = jsonObjects["employees"] as! [[String : AnyObject]]
+                
+                print("org name = \(organizationName)  employess = \(organizationEmployees)")
+                
+                newOrganization = NSEntityDescription.insertNewObject(forEntityName: "FFOrganization", into: DatabaseController.sharedInstance().context) as! FFOrganization
+                
+                newOrganization.name = organizationName
+                
+                for item in organizationEmployees
+                {
+                    let dateFormatter = DateFormatter()
+                    dateFormatter.dateFormat = "yyyy-MM-dd"
+                    
+                    newEmployee = NSEntityDescription.insertNewObject(forEntityName: "FFEmployee", into: DatabaseController.sharedInstance().context) as! FFEmployee
+
+                    newEmployee.firstName = (item["first_name"] as! String)
```

Comment:
as! crashes the application if does not succeed. Use ?? style, e.g. newEmployee.firstName = item["first_name"] as? String ?? ""